basicsr/losses/losses.py

```diff
@@ -354,6 +354,23 @@ def forward(self, input, target_is_real, is_disc=False):
         # loss_weight is always 1.0 for discriminators
         return loss if is_disc else loss * self.loss_weight
 
+    def forward(self, input, target_is_real, is_disc=False):
+        """
+            20210525 [Lotayou]: Add support for multiscale discriminator loss
+            Note the input may not be just one tensor, but a list of tensors
+        """
+        if isinstance(input, list):
+            loss = 0
+            for pred_i in input:
+                if isinstance(pred_i, list):  # in case of multiscale feature matching
+                    pred_i = pred_i[-1]
```

Comment:
It only uses the last element of pred_i?

And could you describe the case when pred_i is a list?

Reply:
We use multiscale discriminator so the output is a list. Moreover, if we want to compute feature matching loss we need to save intermediate feature maps, so the output could be a list of [a list of tensors]. For GANLoss we only need to compute the loss over the last item.